src/main/java/org/stellar/sdk/InvokeHostFunctionOperation.java

```diff
@@ -0,0 +1,153 @@
+package org.stellar.sdk;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Objects;
+import java.util.Arrays;
+import org.stellar.sdk.xdr.*;
+
+/**
+ * Represents <a
+ * href="https://developers.stellar.org/docs/fundamentals-and-concepts/list-of-operations#invoke-host-function"
+ * target="_blank">InvokeHostFunction</a> operation.
+ *
+ * @see <a href="https://developers.stellar.org/docs/fundamentals-and-concepts/list-of-operations"
+ *     target="_blank">List of Operations</a>
+ */
+public class InvokeHostFunctionOperation extends Operation {
+
+  private final HostFunction hostFunction;
+  private SorobanAuthorizationEntry[] auth;
```

Comment:
could leverage generic java collection type instead of array, such as `Collection<SorobanAuthorizationEntry>`

Reply:
recommended to use lombok for this class instead on this comment - https://github.com/stellar/java-stellar-sdk/pull/481/files#r1270963685

Reply:
Fixed in 5838abd4f57793f3e9295ff55817d6d433d1d736

Maybe we can refactor all the code later using lombok?

Reply:
yes, would be good to refactor more of the value objects in general for less boilerplate, probably also evaluate bumping the jvm version up to 11 LTS and dependencies, may need to spin up a few tech debt tickets to track.